### ISSUE TITLE
Library can corrupt data if default charset is not UTF-8

### ISSUE THREAD

**Issue by durera** [ibm-watson-iot/iot-java#86]
The Java Client Library can corrupt the data returned by API calls if the default charset is not UTF-8. For example, if a device definition contains characters that are not in Windows code page Cp1252, and ApiClient.getAllDevices() is called to retrieve the device details, and the default charset is Cp1252, the characters will be corrupted.

The system should handle all Unicode characters without corruption.  The reason for the error is the use of the InputStreamReader(InputStream) constructor in APIClient.java:

```
private String readContent(HttpResponse response, String method)
throws IllegalStateException, IOException {

BufferedReader br = new BufferedReader(new InputStreamReader(response.getEntity().getContent()));
```

This constructor uses the default charset. The code should explicitly state the charset like this:

```
BufferedReader br = new BufferedReader(new InputStreamReader(response.getEntity().getContent(), StandardCharsets.UTF_8));
```

There are further instances of InputStreamReader(InputStream) in AbstractClient.java that should be investigated and if necessary corrected.

**Comment by amitmangalvedkar** [comment]
The code changes have been added sometime back and the fix would go in v0.2.5

**Comment by amitmangalvedkar** [comment]
Waiting for review, so reopened.
